### ISSUE TITLE
Tailwind classes not sorting in PHP code

### ISSUE THREAD

**Issue by zepfietje** [Stillat/blade-parser-typescript#48]
Not sure if this worked in previous versions and this is intended, but I found that Tailwind class sorting via their plugin doesn't work for classes in e.g. `@class`:

Input & output:

```blade
<div @class(['text-center flex'])></div>
```

Expected:

```blade
<div @class(['flex text-center'])></div>
```

**Comment by JohnathonKoster** [comment]
I tested a plain PHP file with equivalent code using Prettier PHP and the Tailwind CSS plugin, and those also do not get sorted.

Additionally, when using the Pint option, all of the inline PHP code is extracted and ran through Laravel Pint instead of prettier (so no prettier plugins would apply, regardless).

**Comment by zepfietje** [comment]
Hmm, I see. Would there be some sort of way to work around this? 🤔
In the Filament codebase we have a looot of Tailwind CSS classes in PHP expressions and I'd love to have to sorted automatically (especially since contributors don't always take sorting into account in PRs).

**Comment by JohnathonKoster** [comment]
Do you have additional examples of where they are being used inside PHP expressions?

A workaround would be something like "extract PHP -> format with pint -> get back results -> parse the PHP -> do something with prettier -> convert it back without messing up the Pint formatting results" (which is a bit...much haha)

**Comment by zepfietje** [comment]
Yeah I understand it would be tricky...

> Do you have additional examples of where they are being used inside PHP expressions?

It's all over the Filament codebase to be honest.
An example would be the button component: https://github.com/filamentphp/filament/blob/2.x/packages/support/resources/views/components/button.blade.php.

**Comment by JohnathonKoster** [comment]
Yea, that's a tough one (and still feels a bit dangerous even in those situations). I can probably get something to work... would you be opposed to having to opt in to this kind of behavior?

Something like surrounding those regions with something like (better name/words to come):

```
{{-- formatter-magic-trick-start --}}

{{-- formatter-magic-trick-end --}}
```

**Comment by zepfietje** [comment]
Nice!
Usually I prefer not having to opt in, but I see how this could potentially be dangerous if enabled by default.

What do you think, @danharrin?

**Comment by danharrin** [comment]
Honestly the codebase is so large that having it opt-in per file or line would be unmanageable IMO

**Comment by zepfietje** [comment]
> Honestly the codebase is so large that having it opt-in per file or line would be unmanageable IMO

Yeah that's what I'm afraid for too.

**Comment by JohnathonKoster** [comment]
Thanks for the feedback! I'll go back to the drawing board and explore having it configurable project-wide via. the root `.blade.format.json` file that controls other settings 👍

**Comment by JohnathonKoster** [comment]
This capability is now generally available starting with version 1.5. **You do not need to opt-in** to this feature for each specific part of the Blade template, and the formatter will utilize sensible defaults to determine when to _not_ run this logic. This feature also works when using Laravel Pint 🙂

The formatter will ignore strings contained in conditions, comparison operations, most assignments, etc. Additionally, it will only update strings inside methods named `class` by default (this is configurable).

Strings will be ignored if any of the following is true for safety reasons:

* The string does not contains a space
* The string contains newline characters
* The string contains any of the following characters: `${"'`
* The string contains the `, ` (comma space) character sequence
* Strings inside `{{-- format-ignore-start --}}{{-- format-ignore-end --}}` regions will **not** be changed

This feature can be fine-tuned quite a bit (but ships with sensible defaults) - you can find out more here:

https://github.com/Stillat/blade-parser-typescript/blob/main/formatter/README.md#class-string-emulation

**Comment by zepfietje** [comment]
Huge! Will try this out now.
